pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -65,8 +65,21 @@ protected static PinotDataBufferMemoryManager getMemoryManager(String consumerDi
    * @return Per-partition consumer's status, which typically includes last consumed message timestamp,
    * latest available upstream offset etc
    */
-  public abstract Map<String, ConsumerPartitionState> getConsumerPartitionState();
+  public abstract Map<String, ConsumerPartitionState> getConsumerPartitionState(boolean skipOffsetLag);
```

Comment:
Is this change related? I feel this is more like a performance optimization to avoid always fetching latest offset?

Reply:
It is related. We don't want to fetch latest offset every time we emit metrics. This would put unusual load on the source broker since we will make #replica calls for each topic partition individually. 
I would think of it more like safety - Pinot should not kill the source broker. :)